### PR TITLE
Provide AWS deployment setup

### PR DESCRIPTION
**Pull request by yasinkalafat**

### CODE REVIEW

File: deployment/app/lib/setup-stack.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+          ],
+        }),
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            'ssm:Get*',
+            'ssm:List*',
+          ],
+          resources: [
+            `arn:aws:ssm:*:*:parameter/sidekick/${process.env.STAGE}/*`
```
Comment (suleymanbarman) [inline deployment/app/lib/setup-stack.ts]: it should not be allowed to access `sidekick` parameters

File: deployment/app/lib/setup-stack.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+          ],
+        }),
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            "secretsmanager:GetSecretValue",
+            "secretsmanager:DescribeSecret",
+          ],
+          resources: [
+            `arn:aws:secretsmanager:*:*:secret:sidekick/${process.env.STAGE}/*`
```
Comment (suleymanbarman) [inline deployment/app/lib/setup-stack.ts]: it should not be allowed to access `sidekick` parameters

File: deployment/app/lib/setup-stack.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            `arn:aws:logs:*:*:log-group:/aws/elasticbeanstalk/sidekick-sandbox-todo-java-${process.env.STAGE}*`,
+          ],
+        }),
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            'sns:Publish',
+          ],
+          resources: [
+            `arn:aws:sns:*:*:sidekick-notification-topic-${process.env.STAGE}`,
```
Comment (suleymanbarman) [inline deployment/app/lib/setup-stack.ts]: it should not be allowed to access `sidekick` resources

File: deployment/app/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "sidekick-api-deploy-app",
```
Comment (suleymanbarman) [inline deployment/app/package.json]: should be updated

File: deployment/app/package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  "version": "0.1.0",
+  "bin": {
+    "app": "bin/app.js"
+  },
+  "private": true,
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/runsidekick/sidekick.git"
```
Comment (suleymanbarman) [inline deployment/app/package.json]: should be updated

File: deployment/lab/.env [hunk 6]
```diff
@@ -0,0 +1,6 @@
+STAGE=lab
+DOMAIN_NAME=runsidekick.me
+SANDBOX_APP_DOMAIN_NAME=todoapp.sandbox.runsidekick.me
```
Comment (suleymanbarman) [inline deployment/lab/.env:4]: it should be `todoapp.sandbox`

Comment (yasinkalafat) [reply deployment/lab/.env:4]: eb-option-settings prepared to work with this config

**Review by rwxdash — APPROVED** [review]

**Pull request merged**